cephfs/admin/clone.go

```diff
@@ -2,6 +2,20 @@
 
 package admin
 
+import (
+	"strings"
+)
+
+const notProtectedSuffix = "is not protected"
+
+// NotProtectedError error values will be returned by CloneSubVolumeSnapshot in
+// the case that the source snapshot needs to be protected but is not.  The
+// requirement for a snapshot to be protected prior to cloning varies by Ceph
+// version.
+type NotProtectedError struct {
```

Comment:
Should `*Error` types not implement an `Error()` function?

Reply:
NotProtectedError will "inherit" the Error function from the embedded response. This exported type can be seen as a special instance of response that callers can check for to handle the varying behavior between different ceph versions.

Reply:
Ah, yes, I missed that!